lib/class-wp-rest-menu-items-controller.php

```diff
@@ -0,0 +1,658 @@
+		}
+
+		if ( in_array( 'menu_order', $fields, true ) ) {
+			$data['menu_order'] = absint( $menu_item->menu_order ); // Same as post_parent, expose as integer.
+		}
+
+		if ( in_array( 'target', $fields, true ) ) {
+			$data['target'] = $menu_item->target;
+		}
+
+		if ( in_array( 'classes', $fields, true ) ) {
+			$data['classes'] = (array) $menu_item->classes;
+		}
+
+		if ( in_array( 'xfn', $fields, true ) ) {
+			$data['xfn'] = (array) $menu_item->xfn;
+		}
+
+		if ( in_array( 'meta', $fields, true ) ) {
+			$data['meta'] = $this->meta->get_value( $menu_item->ID, $request );
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		// Wrap the data in a response object.
+		$response = rest_ensure_response( $data );
+
+		$links = $this->prepare_links( $menu_item );
+		$response->add_links( $links );
+
+		if ( ! empty( $links['self']['href'] ) ) {
+			$actions = $this->get_available_actions( $menu_item, $request );
+
+			$self = $links['self']['href'];
+
+			foreach ( $actions as $rel ) {
+				$response->add_link( $rel, $self );
+			}
+		}
+
+		/**
+		 * Filters the post data for a response.
+		 *
+		 * The dynamic portion of the hook name, `$this->post_type`, refers to the post type slug.
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param object          $post     Post object.
+		 * @param WP_REST_Request  $request  Request object.
+		 */
+		return apply_filters( "rest_prepare_{$this->post_type}", $response, $post, $request );
+	}
+
+	/**
+	 * Retrieves the term's schema, conforming to JSON Schema.
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		$schema = array(
+			'$schema' => 'http://json-schema.org/draft-04/schema#',
+			'title'   => $this->post_type,
+			'type'    => 'object',
+		);
+
+		$schema['properties']['title'] = array(
+			'description' => __( 'The title for the object.' ),
+			'type'        => 'object',
+			'context'     => array( 'view', 'edit', 'embed' ),
+			'arg_options' => array(
+				'sanitize_callback' => null, // Note: sanitization implemented in self::prepare_item_for_database().
+				'validate_callback' => null, // Note: validation implemented in self::prepare_item_for_database().
+			),
+			'properties'  => array(
+				'raw'      => array(
+					'description' => __( 'Title for the object, as it exists in the database.' ),
+					'type'        => 'string',
+					'context'     => array( 'edit' ),
+				),
+				'rendered' => array(
+					'description' => __( 'HTML title for the object, transformed for display.' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit', 'embed' ),
+					'readonly'    => true,
+				),
+			),
+		);
+
+		$schema['properties']['original_title'] = array(
```

Comment:
Can you expand on this? What is the original title for?

Reply:
Original title, is the title of the object in menu. So a term / post etc. While digging around different implementation of menus in core, I found in the customizer class. 
https://github.com/WordPress/WordPress/blob/3e6ed38da102dd4668736a4370c7cbcc7fd050d2/wp-includes/customize/class-wp-customize-nav-menu-item-setting.php#L268-L294

Seemed valid enough, so I added to the endpoint.

Reply:
I think would be better exposed by generating an `embeddable` link to the object for the menu.

Reply:
I like this idea of an embedded link. 
What what about custom links?
What about post type and taxonomy that don't have endpoints?

Reply:
Removed